protocol/app/upgrades/v5.0.0/upgrade.go

```diff
@@ -204,6 +206,39 @@ func initializeOIMFCaps(
 	}
 }
 
+func voteExtensionsUpgrade(
+	ctx sdk.Context,
+	keeper consensusparamkeeper.Keeper,
+) {
+	currentParams, err := keeper.Params(ctx, &consensustypes.QueryParamsRequest{})
+	if err != nil {
+		panic(fmt.Sprintf("failed to retrieve existing consensus params in VE upgrade handler: %s", err))
+	}
+	if currentParams.Params.Abci.VoteExtensionsEnableHeight != 0 {
+		panic(fmt.Sprintf(
+			"unable to update VE Enable Height since its current value of %d is already non-zero",
+			currentParams.Params.Abci.VoteExtensionsEnableHeight))
+	}
+	veEnableHeight := ctx.BlockHeight() + 4
```

Comment:
I think + 1 should actually be fine according to this documentation:
https://docs.cosmos.network/main/build/architecture/adr-064-abci-2.0#upgrading

Reply:
Yeah, I think david wants to get a little bit of separation between when the other upgrades are applied and when the VEs start hitting the chain. I don't really think it makes a different one way or another personally.

Reply:
Agree either could work. For my own understanding, did we remove price daemon in `v5` or is it available as a backup? If the former, does this mean no price update will happen for `deltaBlock`?

Reply:
If you use cosmovisor and your node restarts on the upgrade height it should be running with Prepare/Process Proposal handlers which expect price updates to come from vote extensions. If vote extensions are not yet enabled in the consensus parameters, it will generate an empty `MsgUpdateMarketPrices` and the ProcessProposal handler will only accept an empty message as well.

So to answer your question, yes every block we delay VEs for will have no price updates in it.